### DISCUSSION TITLE
Access app config in markdown content

### DISCUSSION THREAD

**Discussion by julesrx** [nuxt/content#1696] [Q&A] [answered]
Hi,

I'm creating a site using the document-driven mode, and I would like to access the app config data in the markdown.
I'm struggling to understand how to do this properly.

Currently I created a component (in `components/content`) and inserting in the markdown, but this seems strange just for static content generated during the `nuxt generate` command.
```vue
<script setup lang="ts">
const config = useAppConfig();
</script>

<template>
  <span>{{ config.title }}</span>
</template>
```

```md
**<site-title></site-title>**  
```

I saw that when I use the `{{ ... }}` syntax in the markdown, the renderer seems to render data, but I don't understand where this data comes from, so this just results in `null`.

I would like to do something like this :
```md
{{ appConfig.title }}
```

Is there any way to do this, without having to create individual components ?

**Comment by ManasMadrecha** [comment, marked as answer]
Just create one global component, and pass the key you want to display as a prop. So you can easily use that one component in your markdown files, to display any app config value. 

Another solution: in the `content after parse` hook, add the keys you want to use in your markdown files as the autogenerated frontmatter. Then you can use the `{{}}` syntax to display any such key.

**Reply by julesrx** [reply]
I ended up using global components.
Thanks 👌
